test-app/src/main/java/org/readium/r2/testapp/outline/NavigationFragment.kt

```diff
@@ -83,13 +83,7 @@ class NavigationFragment : Fragment() {
     }
 
     private fun onLinkSelected(link: Link) {
-        val locator = link.toLocator().let {
-            // progression is mandatory in some contexts
-            if (it.locations.fragments.isEmpty())
-                it.copyWithLocations(progression = 0.0)
-            else
-                it
```

Comment:
I made sure that `locatorFromLink()` returns a `Locator` with either `progression` or `fragments` set.